fe/fe-core/src/main/java/com/starrocks/external/hive/CachingHiveMetastore.java

```diff
@@ -0,0 +1,287 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.external.hive;
+
+import com.google.common.base.Preconditions;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Streams;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.HiveMetaStoreTable;
+import com.starrocks.catalog.Table;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Throwables.throwIfInstanceOf;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static com.google.common.cache.CacheLoader.asyncReloading;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.util.concurrent.MoreExecutors.newDirectExecutorService;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class CachingHiveMetastore implements IHiveMetastore {
+    public static final long NEVER_CACHE = 0;
+    public static final long NEVER_EVICT = -1;
+    public static final long NEVER_REFRESH = -1;
+    protected final IHiveMetastore metastore;
+
+    protected LoadingCache<String, List<String>> databaseNamesCache;
+    protected LoadingCache<String, List<String>> tableNamesCache;
+    protected LoadingCache<HiveTableName, List<String>> partitionKeysCache;
+
+    protected LoadingCache<String, Database> databaseCache;
+    protected LoadingCache<HiveTableName, Table> tableCache;
+
+    protected LoadingCache<NewHivePartitionName, Partition> partitionCache;
```

Comment:
add comment

Reply:
Others could view the definition of class. I think it isn't necessary to write too many comments to introduce these on here